rtloader/common/builtins/datadog_agent.c

```diff
@@ -349,6 +357,41 @@ static PyObject *log_message(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+/*! \fn PyObject *set_check_metadata(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.set_check_metadata` method, updating
+    the value in the cache.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a 3-ary tuple containing the unique ID of a check
+    instance, the name of the metadata entry, and the value of said entry.
+    \return A PyObject* pointer to `None`.
+
+    This function is callable as the `datadog_agent.set_check_metadata` Python method and
+    uses the `cb_set_check_metadata()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *set_check_metadata(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_set_check_metadata == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    char *check_id, *name, *value;
+
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    // datadog_agent.set_check_metadata(check_id, name, value)
+    if (!PyArg_ParseTuple(args, "sss", &check_id, &name, &value)) {
```

Comment:
Don't we know the check_id already, instead of it being passed from the check itself?

Reply:
I don't think we do. You see a way?